src/framework/components/element/image-element.js

```diff
@@ -519,36 +525,26 @@ Object.assign(pc, function () {
                     }
                 }
 
-                this._uvs[0] = rect.x / w;
-                this._uvs[1] = rect.y / h;
-                this._uvs[2] = (rect.x + rect.z) / w;
-                this._uvs[3] = rect.y / h;
-                this._uvs[4] = (rect.x + rect.z) / w;
-                this._uvs[5] = (rect.y + rect.w) / h;
-                this._uvs[6] = rect.x / w;
-                this._uvs[7] = (rect.y + rect.w) / h;
+                // Update vertex texture coordinates
+                vertexDataF32[6] = rect.x / w;
+                vertexDataF32[7] = rect.y / h;
+                vertexDataF32[14] = (rect.x + rect.z) / w;
+                vertexDataF32[15] = rect.y / h;
+                vertexDataF32[22] = (rect.x + rect.z) / w;
+                vertexDataF32[23] = (rect.y + rect.w) / h;
+                vertexDataF32[30] = rect.x / w;
+                vertexDataF32[31] = (rect.y + rect.w) / h;
 
-                var vb = mesh.vertexBuffer;
-                var it = new pc.VertexIterator(vb);
-                var numVertices = 4;
-                for (i = 0; i < numVertices; i++) {
-                    it.element[pc.SEMANTIC_POSITION].set(this._positions[i * 3 + 0], this._positions[i * 3 + 1], this._positions[i * 3 + 2]);
-                    it.element[pc.SEMANTIC_NORMAL].set(this._normals[i * 3 + 0], this._normals[i * 3 + 1], this._normals[i * 3 + 2]);
-                    it.element[pc.SEMANTIC_TEXCOORD0].set(this._uvs[i * 2 + 0], this._uvs[i * 2 + 1]);
-
-                    it.next();
-                }
-                it.end();
+                vb.unlock();
 
-                mesh.aabb.compute(this._positions);
+                mesh.aabb.setMinMax(pc.Vec3.ZERO, new pc.Vec3(w, h, 0));
```

Comment:
Is this right? Seems like you're BB will always have one corner at [0,0,0]?

Reply:
Well spotted. Fixed.